cue/dac-utils/panel-group.cue

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This package offers an utility to build a simple, standard layout easily,
+// without having to manipulate directly the {x,y,w,h} coordinates of each
+// and every panel.
+
+package panelGroup
+
+import (
+	"math"
+	"github.com/perses/perses/pkg/model/api/v1"
+	v1Dashboard "github.com/perses/perses/pkg/model/api/v1/dashboard"
+)
+
+// expected user inputs
+#panels: [string]: v1.#Panel
+#title: string
+#cols: >0 & <=#gridCols
+#height: number | *6
+
+// intermediary compute
+#gridCols: 24
+#panelsAsList: [for k, p in #panels {p, name: k}]
+#width: math.Trunc(#gridCols / #cols)
+#panelsX: [for i, _ in #panelsAsList {
+    #width * math.Round(math.Mod(i, #cols))
+}]
+#panelsY: [for i, _ in #panelsAsList {
+    #height * math.Trunc(i / #cols)
+}]
+
+// output: the final layout, in the format expected by the Perses dashboard.
+v1Dashboard.#Layout & {
```

Comment:
I think a docs on each utilities method/function you are providing would help a lot to have more inside about what we are providing.

Probably a doc for each files and a sort of scenario to gives a context why we are providing such utilities method/function

Reply:
For such lib doc useful to developers, I prefer to provide "embedded" doc (ideally jsdoc style that could be rendered in a webpage) rather than external markdown files, that have a higher risk to later get desynchronized with the actual code imho, hence why I put comments in each file & a complete example rather than documented these in the user guide. Still yes for now it's only few lines of text and no fancy jsdoc-style content so I'll try to enrich this part.

Reply:
After discussing this live, 'agreed that we could have the best of both worlds here: dev doc in a jsdoc way that could be used to generate markdown files, to be integrated in the perses website. That's work for a future PR though :D